### PR TITLE
WT-2667 Split evergreen compile-posix into separate compile and test …

### PR DESCRIPTION
**Pull request by daveh86**
…stages

### CODE REVIEW

**Comment by daveh86** [comment]
@agorrod, mind taking a look at this?

File: test/mciproject.yml [hunk 1]
```diff
@@ -21,25 +33,53 @@ post:
...
             set -o errexit
             set -o verbose
 
-            ./build_posix/reconf
-            ${configure_env_vars|} ./configure --enable-diagnostic --enable-python --enable-zlib
+            ./reconf
+            ${configure_env_vars|} ../configure --enable-diagnostic --enable-python --enable-zlib
             ${make_command|make} ${smp_command|}
             ${make_command|make} check
+      - command: shell.exec
```
Comment (markbenvenuto) [inline test/mciproject.yml]: use `archive.targz_pack` instead, see https://github.com/mongodb/mongo/blob/master/etc/evergreen.yml for examples.

File: test/mciproject.yml [hunk 2]
```diff
@@ -8,6 +8,18 @@ functions:
...
+      params:
+        aws_key: ${aws_key}
+        aws_secret: ${aws_secret}
+        remote_file: wiredtiger/${build_variant}/${revision}/artifacts/${build_id}.tgz
+        local_file: wiredtiger.tgz
+        bucket: build_external
+    - command: shell.exec
+      params:
+        script: |
+          tar -xvf wiredtiger.tgz
```
Comment (markbenvenuto) [inline test/mciproject.yml]: Don't you need `z`?

Comment (daveh86) [inline test/mciproject.yml]: Tar wierdness, the {{z}} is implicit. However, you're right and I should be explicit.

File: test/mciproject.yml (hunk 1 above)
Comment (daveh86) [inline test/mciproject.yml]: I tried that, but the wildcard didn't work. Gave it a second go this evening with double wildcard and that seems to be the key.

**Comment by daveh86** [comment]
@agorrod, can you have a look at this?

**Comment by markbenvenuto** [comment]
@daveh86 I like the `-j` flag on the tests, nice.

BTW, you can also exclude files (like say `.git`).

LGTM

File: test/mciproject.yml [hunk 3]
```diff
@@ -21,25 +29,55 @@ post:
...
-          working_dir: "wiredtiger"
+          working_dir: "wiredtiger/build_posix"
           script: |
             set -o errexit
             set -o verbose
 
-            ./build_posix/reconf
-            ${configure_env_vars|} ./configure --enable-diagnostic --enable-python --enable-zlib
+            ./reconf
+            ${configure_env_vars|} ../configure --enable-diagnostic --enable-python --enable-zlib
```
Comment (agorrod) [inline test/mciproject.yml]: @daveh86 Can you add the ``--enable-strict`` option to the command line?

**Comment by agorrod** [comment]
Thanks @daveh86 I had one request, apart from that this lgtm - it's a big step forward in our Evergreen testing.

**Comment by daveh86** [comment]
@agorrod, done.

**Comment by agorrod** [comment]
Thanks @daveh86 lgtm - I'll merge.

**Pull request merged**